tensornetwork/backends/jax/jax_backend.py

```diff
@@ -433,12 +431,160 @@ def A(H,x):
           type(initial_state)))
     if A not in _CACHED_MATVECS:
       _CACHED_MATVECS[A] = libjax.tree_util.Partial(A)
-    if not hasattr(self, '_jaxlan'):
-      # pylint: disable=attribute-defined-outside-init
-      self._jaxlan = jitted_functions._generate_jitted_eigsh_lanczos(libjax)
+    eigsh_lanczos = jitted_functions._generate_jitted_eigsh_lanczos(libjax)
+    return eigsh_lanczos(_CACHED_MATVECS[A], args, initial_state,
+                         num_krylov_vecs, numeig, delta, reorthogonalize)
+
+  def gmres(self,
+            A_mv: Callable,
+            b: Tensor,
+            A_args: Optional[List] = None,
+            A_kwargs: Optional[dict] = None,
+            x0: Optional[Tensor] = None,
+            tol: float = 1E-05,
+            atol: Optional[float] = None,
+            num_krylov_vectors: Optional[int] = None,
+            maxiter: Optional[int] = 1,
```

Comment:
remove `Optional`

Reply:
The Optional is needed so that the default can be b.size (instead of a fixed number)

Reply:
isn't the default 1?